synapse/events/builder.py

```diff
@@ -128,14 +128,12 @@ async def build(
             )
 
         format_version = self.room_version.event_format
+        # The types of auth/prev events changes between event versions.
+        prev_events: Union[List[str], List[Tuple[str, Dict[str, str]]]]
+        auth_events: Union[List[str], List[Tuple[str, Dict[str, str]]]]
         if format_version == EventFormatVersions.V1:
-            # The types of auth/prev events changes between event versions.
-            auth_events: Union[
-                List[str], List[Tuple[str, Dict[str, str]]]
-            ] = await self._store.add_event_hashes(auth_event_ids)
-            prev_events: Union[
-                List[str], List[Tuple[str, Dict[str, str]]]
-            ] = await self._store.add_event_hashes(prev_event_ids)
+            auth_events = await self._store.add_event_hashes(auth_event_ids)
+            prev_events = await self._store.add_event_hashes(prev_event_ids)
```

Comment:
Is this changing anything or just moving the types up?

Reply:
Just moving the types up. I was confused because `add_event_hashes` returns the `List[Tuple[...]]`, which I thought made the union unnecessary. But in the `else` branch, both event lists are `List[str]`. I hoped that moving it up would make this clearer.